lib/logstash/codecs/telemetry.rb

```diff
@@ -258,26 +345,75 @@ def ts_codec_extract_path_key_value(path, data, filter_table, depth)
     end # branch, v iteration over hash passed down
   end # ts_codec_extract_path_key_value
 
+  private
+  def decode_kv_table(table,evs)
+
+    ev = Hash.new
+
+    if table[:timestamp]
```

Comment:
In the case timestamp is 0 you will need to inherit timestamp from above layer (i.e. passed down from top level or below). The way it works is timestamps are propagated down unless a more specific timestamp is available.